File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {
    
    seen := make(map[int]int)
    for i , n := range nums { 
        if  _ , ok := seen[n] ; ok && i-seen[n] <=k{ 
            return true 
        }
        seen[n]  = i 
        
    }    
    return false 
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        
        unordered_set<int> set ;
        for(int i=0 ; i < nums.size() ; i++){
            if ( i > k) set.erase(nums[i-k-1]) ;
            
            if(!set.insert(nums[i]).second){
                return true; 
            }
        }
        return false; 
        
        
        
    }
};